src/wdb/QueryRunner.h

```diff
@@ -91,24 +87,16 @@ class QueryRunner : public wbase::TaskQueryRunner, public std::enable_shared_fro
 
 private:
     bool _initConnection();
-    void _setDb();
 
     /// Dispatch with output sent through a SendChannel
     bool _dispatchChannel();
     MYSQL_RES* _primeResult(std::string const& query);  ///< Obtain a result handle for a query.
 
-    static size_t _getDesiredLimit();
-
     wbase::Task::Ptr const _task;  ///< Actual task
 
-    qmeta::CzarId _czarId = 0;  ///< To be replaced with the czarId of the requesting czar.
-
     /// Resource reservation
     ChunkResourceMgr::Ptr _chunkResourceMgr;
-    std::string _dbName;
     std::atomic<bool> _cancelled{false};
-    std::weak_ptr<xrdsvc::StreamBuffer> _streamBuf;  ///< used release condition variable on cancel.
```

Comment:
As long as `this` is valid, locking `_streamBuf` should be valid. It lock should return nullptr if it hasn't been set. I suspect the problem was caused by the other memory issues we have seen.

Reply:
I agree with you. I too was surprised when I noticed this crash and a huge reference counter on the pointer. In any case, it was dead code that had to be eliminated.